packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts

```diff
@@ -482,50 +477,61 @@ export const splitFormatIntoSections = <TDate>(
     return null;
   };
 
-  const splitFormat = (token: string) => {
-    const escapedParts = getEscapedPartsFromFormat(utils, token);
-    let currentTokenValue = '';
-
-    for (let i = 0; i < token.length; i += 1) {
-      const escapedPartOfCurrentChar = escapedParts.find(
-        (escapeIndex) => escapeIndex.start <= i && escapeIndex.end >= i,
+  // Expand the provided format
+  let formatExpensionOverflow = 10;
+  let prevFormat = format;
+  let nextFormat = utils.expandFormat(format);
+  while (nextFormat !== prevFormat) {
+    prevFormat = nextFormat;
+    nextFormat = utils.expandFormat(prevFormat);
+    formatExpensionOverflow -= 1;
+    if (formatExpensionOverflow < 0) {
+      throw new Error(
+        'MUI: The format exmpension seems to be  enter in an infint loop. Please open an issue with the format passed to the picker component',
       );
+    }
+  }
+  const expandedFormat = nextFormat;
+
+  // Get start/end indexes of escaped sections
+  const escapedParts = getEscapedPartsFromFormat(utils, expandedFormat);
+
+  // This RegExp test if the begining of a string correspond to a supported token
+  const isTokenStartRegExp = new RegExp(`^(${Object.keys(utils.formatTokenMap).join('|')})`);
 
-      const char = token[i];
-      const isEscapedChar = escapedPartOfCurrentChar != null;
-
-      if (!isEscapedChar && char.match(/([A-Za-z]+)/)) {
-        currentTokenValue += char;
-      } else {
-        // If we are on the opening or closing character of an escaped part of the format,
-        // Then we ignore this character.
-        const isEscapeBoundary =
-          (isEscapedChar && escapedPartOfCurrentChar?.start === i) ||
-          escapedPartOfCurrentChar?.end === i;
-
-        if (!isEscapeBoundary) {
-          const expandedToken = commitToken(currentTokenValue);
-          if (expandedToken != null) {
-            splitFormat(expandedToken);
-          }
-
-          currentTokenValue = '';
-          if (sections.length === 0) {
-            startSeparator += char;
-          } else {
-            sections[sections.length - 1].endSeparator += char;
-          }
+  let currentTokenValue = '';
+
+  for (let i = 0; i < expandedFormat.length; i += 1) {
+    const escapedPartOfCurrentChar = escapedParts.find(
+      (escapeIndex) => escapeIndex.start <= i && escapeIndex.end >= i,
+    );
+
+    const char = expandedFormat[i];
+    const isEscapedChar = escapedPartOfCurrentChar != null;
+    const potentialToken = `${currentTokenValue}${expandedFormat.slice(i)}`;
+    if (!isEscapedChar && char.match(/([A-Za-z]+)/) && isTokenStartRegExp.test(potentialToken)) {
```

Comment:
Can we drop the old regexp here ?
For me it's a weaker test than `isTokenStartRegExp` so it's redundant

```suggestion
    if (!isEscapedChar && isTokenStartRegExp.test(potentialToken)) {
```

Reply:
In fact, it's not the same.
- `isTokenStartRegExp.test(potentialToken)` verifies that we are at the beginning of a token
- `char.match(/([A-Za-z]+)/)` verify we are still adding a character to the token.

So for example with `hh:mm` all the substrings (`h`, `hh`, `hh:`, and `hh:m`) pass the first regexp because they all start by `h` which is a valid token. It's stopped by the second regexp when matching `:` which is not a letter

This works because we can assume all tokens are delimited by a non-letter characters, at least space.